src/nodes/math/HashNode.js

```diff
@@ -1,34 +1,15 @@
-import Node, { addNodeClass } from '../core/Node.js';
-import { addNodeElement, nodeProxy } from '../shadernode/ShaderNode.js';
+import { Fn, addNodeElement } from '../shadernode/ShaderNode.js';
 
-class HashNode extends Node {
+export const hash = Fn( ( [ seed ] ) => {
 
-	constructor( seedNode ) {
+	// Taken from https://www.shadertoy.com/view/XlGcRh, originally from pcg-random.org
 
-		super();
+	const state = seed.toUint().mul( 747796405 ).add( 2891336453 );
+	const word = state.shiftRight( state.shiftRight( 28 ).add( 4 ) ).bitXor( state ).mul( 277803737 );
+	const result = word.shiftRight( 22 ).bitXor( word );
 
-		this.seedNode = seedNode;
+	return result.toFloat().mul( 1 / 2 ** 32 ); // Convert to range [0, 1)
 
-	}
-
-	setup( /*builder*/ ) {
-
-		// Taken from https://www.shadertoy.com/view/XlGcRh, originally from pcg-random.org
-
-		const state = this.seedNode.toUint().mul( 747796405 ).add( 2891336453 );
-		const word = state.shiftRight( state.shiftRight( 28 ).add( 4 ) ).bitXor( state ).mul( 277803737 );
-		const result = word.shiftRight( 22 ).bitXor( word );
-
-		return result.toFloat().mul( 1 / 2 ** 32 ); // Convert to range [0, 1)
-
-	}
-
-}
-
-export default HashNode;
-
-export const hash = nodeProxy( HashNode );
+} );
 
 addNodeElement( 'hash', hash );
-
-addNodeClass( 'HashNode', HashNode );
```

Comment:
If the file does a `Fn` instead of a `Node`, should the file be renamed to `HashFn.js`?

Reply:
What do you think just `Hash.js` like [TriNoise3D.js](https://github.com/mrdoob/three.js/blob/dev/src/nodes/math/TriNoise3D.js)?

Reply:
That works too I think 👌